libtiledbvcf/src/dataset/attribute_buffer_set.cc

```diff
@@ -22,8 +22,10 @@ void AttributeBufferSet::allocate_fixed(
 
   // Requesting 0 MB will result in a 1 KB allocation. This is used by the
   // tests to test the path of incomplete TileDB queries.
-  nbytes = mem_budget_mb == 0 ? 1024 : (mem_budget_mb * 1024 * 1024);
-  num_offsets = nbytes / sizeof(uint64_t);
+  if (mem_budget_mb == 0) {
```

Comment:
See line 20 for context. The issue is that the correct allocation is set on line 20, but we're overwriting it here (line 25 on the original) with the incorrect allocation by not dividing by num_buffers.